types/data.d.ts

```diff
@@ -43,7 +43,9 @@ export const enum Event {
     Paint = 35,
     ContentfulPaint = 36,
     Memory = 37,
-    Connection = 38
+    Connection = 38,
+    Backup = 39,
```

Comment:
Briefly what are the semantics of these 2 new events (backup and upgrade)?

Reply:
Backup => It backs up any layout events in lean mode until clarity.upgrade() is called. So instead of throwing away layout events like before, now we store them in memory. 

Upgrade => This event retrieves stored events in backup and sends them to server. When this happens, lean mode  (boxmodel like) page seamless upgrades into full blow session from the beginning of this page start time.

---
In reply to: [368666638](https://github.com/microsoft/clarity-js/pull/166#discussion_r368666638) [](ancestors = 368666638)

Reply:
Add comment please with above. This is as good a place as any for the comment. 

Reading above I'm still not clear:

* What exactly does a backup event represent and is it internal or is something sent with data? If so, when is the event sent and what data does it contain? I assume it doesn't contain
* Sam question for the upgrade event. What you are describing above is the behavior of the client code, not the event the data semantics.  Events don't *do* anything they represent something happening. 
* It might be good to bundle these "client state transition" events into a common event as each kind of event ends up having overhead to add. Judgment call. 

---
In reply to: [368805833](https://github.com/microsoft/clarity-js/pull/166#discussion_r368805833) [](ancestors = 368805833,368666638)